src/serializer.rs

```diff
@@ -50,15 +52,20 @@ fn write_numeric<W>(value: f32, int_value: Option<i32>, has_sign: bool, dest: &m
         dest.write_str("+")?;
     }
 
-    if value == 0.0 && value.is_sign_negative() {
+    let result = if value == 0.0 && value.is_sign_negative() {
         // Negative zero. Work around #20596.
-        dest.write_str("-0")?
+        Cow::Borrowed("-0")
     } else {
-        write!(dest, "{}", value)?
-    }
+        let mut result = String::new();
+        dtoa_short::write(&mut result, value)?;
```

Comment:
Unless you need to extend and existing value, out parameters are not idiomatic in Rust and can be replaced by returning a tuple. Or this case, replacing `Result<(), Error>` with `Result<bool, Error>`, or maybe a custom enum instead of `bool`.